lib/grizzly/application.ex

```diff
@@ -65,6 +66,47 @@ defmodule Grizzly.Application do
     end
   end
 
+  defp check_for_tuntap({:unix, :darwin}) do
+    System.cmd("kextstat", [ "-b", "net.sf.tuntaposx.tap" ])
+    |> elem(0)
+    |> String.contains?("net.sf.tuntaposx.tap")
+    |> case do: (
+      true ->
+        ; # all OK
+      _ ->
+        msg = """
+        The kernel extension tuntab does not appear to be loaded. You
+        can install it using `brew cask install tuntap`.
+        """
+        Logger.error(msg)
+        raise(msg)
+    )
+  end
+
+  defp find_zip_gateway do
+    path = Application.get_env(:grizzly, :zipgateway_path, "/usr/sbin/zipgateway")
+    case File.stat(path) do
+      { :error, posix } ->
+        msg = """
+        Cannot find the zipgateway executable (looked for it at #{inspect path}.
+
+        If it is located somewhere else, please update the config:
+
+            config :grizzly,
+              zipgateway_path: "«path»"
+        """
+        Logger.error(msg)
+        raise(msg)
+      { :ok, _stat } ->
+        # could check the mode, but not really worth it
+        path
+    end
+  end
+
+  defp check_for_tuntap({:unix, _}) do
```

Comment:
Can we move this pattern match up to be under the `check_for_tuntap` function?

Reply:
both done...